moshi/src/main/java/com/squareup/moshi/JsonAdapter.java

```diff
@@ -30,13 +30,13 @@
  * Converts Java values to JSON, and JSON values to Java.
  */
 public abstract class JsonAdapter<T> {
-  public abstract @Nullable T fromJson(JsonReader reader) throws IOException;
+  @CheckReturnValue public abstract @Nullable T fromJson(JsonReader reader) throws IOException;
```

Comment:
There may be some use case for having a delegate adapter that throws away a value you don't care about, but that seems weird enough that a suppression in app code with an explanation is warranted.

Reply:
definitely. `reader.skipValue()` is better.